### PR TITLE
V1

### PR DESCRIPTION
**Pull request by andyrichardson**
Once changes have passed review, we can go ahead and merge this into master.

There will be some documentation additions and minor optimisations required before this is ready for release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,49 +138,46 @@ You can also use functional child style:
...
+- `error` - If there is an error returned when making the query, instead of data, you get this and you can handle it or show a `refetch` button or cry or whatever you wanna do.
+- `data` - This is where your data lives. Once the query returns, This would look like `{ todos: [...] }`.
+- `mutations` - This is a collection of functions for triggering a mutation, each key matching the mutations you just provided to the _Connect_ component. To call the _addTodo_ mutation with variables, you would do `mutations.addTodo(myVars)`.
+- `refetch` - This is a method that you can use to manually refetch your query. You can skip the cache, hit the server and repopulate the cache by calling this like `refetch(true)`.
 
-So why do we use these `query` and `mutation` functions before passing them? Variables, thats why. If you wanted to pass a query with variables, you would construct it like so:
+So why do we use these `createQuery` and `createMutation` functions before passing them? Variables, thats why. If you wanted to pass a query with variables, you would construct it like so:
```
Comment (lukejacksonn) [inline README.md]: that's ?
```suggestion
So why do we use these `createQuery` and `createMutation` functions before passing them? Variables, that's why. If you wanted to pass a query with variables, you would construct it like so:
```

File: src/components/__snapshots__/provider.test.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          "getSnapshotBeforeUpdate": true,
+          "setState": Object {
+            "skipsComponentDidUpdateOnNullish": true,
+          },
+        },
+      },
+    },
+  },
+}
+`;
```
Comment (blainekasten) [inline src/components/__snapshots__/provider.test.tsx.snap]: protip. You should add either `jest-enzyme` or `enzyme-to-json` to this branch. Right now the serialized snapshots from enzyme include a bunch of enzyme internals. So any update to enzyme could break all of these.

Comment (andyrichardson) [reply src/components/__snapshots__/provider.test.tsx.snap]: Thanks @blainekasten - I've opened an issue 👍

**Comment by kitten** [comment]
Left over todos:

- [ ] Rename `ExchangeResult` to `OperationResult`
- [ ] Check / review recent changes
- [ ] Check examples and remove `example/`
- [ ] Split `src/react/` to `src/components/` and `src/hooks/`

**Comment by kitten** [comment]
Merging since everything seems :100: 👌 and we can mostly iron out some things on `master`, update/write docs separately, and fix a couple of things later, maybe even after v1. This branch has gotten oooooold 😛 

Long story short:

**Comment by lukejacksonn** [comment]
Well done @andyrichardson et al. This was one chonky PR. Good work 🎉

**Pull request merged**